File: core/src/dc/targetman/gamelogic/VitalLimbsSystem.java
package dc.targetman.gamelogic;

import java.util.List;

import dc.targetman.epf.parts.VitalLimbsPart;
import dclib.epf.Entity;
import dclib.epf.EntityManager;
import dclib.epf.EntitySystem;
import dclib.limb.Limb;
import dclib.limb.LimbUtils;

public final class VitalLimbsSystem extends EntitySystem {

	private final EntityManager entityManager;

	public VitalLimbsSystem(final EntityManager entityManager) {
		super(entityManager);
		this.entityManager = entityManager;
	}

	@Override
	protected final void update(final float delta, final Entity entity) {
		VitalLimbsPart vitalLimbsPart = entity.tryGet(VitalLimbsPart.class);
		if (vitalLimbsPart != null) {
			List<Entity> entities = entityManager.getAll();
			for (Limb limb : vitalLimbsPart.getVitalLimbs()) {
				if (LimbUtils.findEntity(entities, limb) == null) {
					entityManager.remove(entity);
					break;
				}
			}
		}
	}

}

File: core/src/main/dc/targetman/character/VitalLimbsSystem.kt
package dc.targetman.character

import dc.targetman.epf.parts.SkeletonPart
import dc.targetman.epf.parts.VitalLimbsPart
import dclib.epf.Entity
import dclib.epf.EntityManager
import dclib.epf.EntitySystem

class VitalLimbsSystem(private val entityManager: EntityManager) : EntitySystem(entityManager) {
    override fun update(delta: Float, entity: Entity) {
        val vitalLimbsPart = entity.tryGet(VitalLimbsPart::class)
        if (vitalLimbsPart != null) {
            val skeletonPart = entity[SkeletonPart::class]
            val limbs = skeletonPart.getLimbs()
            val isVitalLimbDead = vitalLimbsPart.limbNames.any { limbs.none { it.name == it.name } }
            if (isVitalLimbDead) {
                entityManager.destroy(entity)
            }
        }
    }
}